### PR TITLE
patch/calc_patch: fix path prefix false-positive in _strictly_under

### PR DESCRIPTION
**Pull request by Dev-X25874**
## :rocket: What

Fixes a bug in `_strictly_under` where a raw string `startswith` check
incorrectly matched paths that share a name prefix with a watched
directory but are not actually inside it.

For example, with the default `model_module_dir = "model"`, a file at
`model_weights/checkpoint.bin` would pass the `path.startswith("model")`
guard and be misclassified as a model-code file, generating a spurious
`ModelCodePatch` for a file that lives outside the model directory.

## :computer: How

Changed the prefix check from `path.startswith(dir_path)` to
`path.startswith(dir_path + "/")`, anchoring the match to a directory
boundary. The now-redundant `not path == dir_path` guard was removed,
since `"model/"` can never equal `"model"`.

## :microscope: Testing

- Manually verified the false-positive scenario: `"model_weights/checkpoint.bin"` no longer matches against `"model"`.
- Verified true-positive still works: `"model/foo.py"` correctly matches against `"model"`.
- Verified same-path edge case: `"model"` does not match against `"model"` (unchanged behaviour).
- No existing tests cover `_strictly_under` directly; the fix is a one-line change with no side effects on the call sites.

### CODE REVIEW

File: truss/truss_handle/patch/calc_patch.py [hunk 1]
```diff
@@ -491,6 +491,6 @@ def _strictly_under(path: str, parent_paths: List[str]) -> bool:
     them. Assumes that parent paths themselves are not under each other.
     """
     for dir_path in parent_paths:
-        if path.startswith(dir_path) and not path == dir_path:
+        if path.startswith(dir_path + "/"):
```
Comment (cretz) [inline truss/truss_handle/patch/calc_patch.py]: Unfortunately I am afraid this will break Windows whereas the incorrect prefix match from before did work (but it was incorrect). Maybe:

```suggestion
        if path.startswith(dir_path + os.sep):
```

**Review by cretz — COMMENTED** [review]
Thanks for the contribution! One minor suggestion

File: truss/truss_handle/patch/calc_patch.py (hunk 1 above)
Comment (Dev-X25874) [reply truss/truss_handle/patch/calc_patch.py]: Done — switched to os.sep and added import os. Thanks for the catch!

**Review by cretz — APPROVED** [review]
Thanks for the contribution!
